component-samples/demo/scripts/fdo_toolkit.sh

```diff
@@ -0,0 +1,282 @@
+#
+#
+#  aio_e2e_setup
+#
+#  
+#  Pending Features:
+#  
+#
+#  1. Ability to update rvinfo
+#  2. Ability to update Owner redirect
+#  3. Starting individual services
+#  4. Handling proxy 
+#  5. Add your ideas here
+#
+#
+#
+#
+#
+#
+#################    Documentation   #############################
+
+
+FDO_RELEASE="1.1.5.1"
+
+
+chk() {
+    local exitCode=$1
+    local task=$2
+    if [[ $exitCode == 0 ]]; then return; fi
+    echo "Error: exit code $exitCode from: $task"
+    exit $exitCode
+}
+
+clean_build() {
+  sudo rm -rf  ~/pri_code_base/
+}
+
+download_and_unpack_binaries() {
+
+
+    mkdir -p ~/pri_code_base
+    cd ~/pri_code_base
+    echo "Downloading Protocol Reference Implementation $FDO_RELEASE"
+    curl --progress-bar -LO https://github.com/fido-device-onboard/release-fidoiot/releases/download/v$FDO_RELEASE/pri-fidoiot-v$FDO_RELEASE.tar.gz
+    chk $? 'downloading pri'
+    echo "Unpacking Protocol Reference Implementation $FDO_RELEASE"
+    tar -zxf pri-fidoiot-v$FDO_RELEASE.tar.gz
+    chk $? 'unpacking pri'
+
+}
+
+docker_cleaner() {
+    echo "Stopping all containers"  
+    docker stop $(docker ps -a -q)
+    echo "Stopped all containers"
+}
+
+fdo_docker_cleaner() {
+    echo "Stopping all FDO containers"
+    docker stop pri-fdo-aio pri-fdo-mfg pri-fdo-rv pri-fdo-owner db_fdo-db_1
+    echo "Stopped all FDO containers"
+}
+
+docker_system_pruner() {
+    docker_cleaner
+    echo "Removing all containers, images and volumes"
+    echo "y" | docker system prune -a --volumes
+    echo "Removed all containers, images and volumes"
+}
+
+isDockerComposeAtLeast() {
+    : ${1:?}
+    local minVersion=$1
+    if ! command -v docker-compose >/dev/null 2>&1; then
+        return 1   # it is not even installed
+    fi
+    # docker-compose is installed, check its version
+    lowerVersion=$(echo -e "$(docker-compose version --short)\n$minVersion" | sort -V | head -n1)
+    if [[ $lowerVersion == $minVersion ]]; then
+        return 0   # the installed version was >= minVersion
+    else
+        return 1
+    fi
+}
+
+install_dependencies() {
+
+echo "Installing all FDO dependencies"
+
+if java -version 2>&1 | grep version | grep -q 11.; then
+        echo "Found java 11"
+else
+        echo "Java 11 not found, installing it..."
+        apt-get update && apt-get install -y openjdk-11-jre-headless
+        chk $? 'installing java 11'
+fi
+
+
+if ! dpkg-query -s haveged > /dev/null 2>&1; then
+    echo "Haveged is required, installing it"
+    sudo apt-get install -y haveged
+    chk $? 'installing haveged'
+fi
+
+# If docker isn't installed, do that
+if ! command -v docker >/dev/null 2>&1; then
+    echo "Docker is required, installing it..."
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    chk $? 'adding docker repository key'
+    add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
```

Comment:
On Fresh VM , Docker Installation was failing with error:
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
curl: (28) Failed to connect to download.docker.com port 443 after 278859 ms: Connection timed out
gpg: no valid OpenPGP data found.
Error: exit code 2 from: adding docker repository key